guide/src/types.md

```diff
@@ -108,9 +111,12 @@ lifetime just like `PyAny`.
 the GIL.  Like `PyAny`, this is the most convenient form to use for function
 arguments and intermediate values.
 
+These types all implement `Deref<Target = PyAny>`, so they all expose the same
+methods which can be found on `PyAny`.
+
 **Conversions:**
 
-- To `PyAny`: `obj.as_ref()`
+- To `PyAny`: `obj.as_ref()` or `obj.deref()`
```

Comment:
Since [clippy dislike explicit deref](https://rust-lang.github.io/rust-clippy/master/index.html#explicit_deref_methods), `&*obj` is better?

Reply:
Hmm it'll actually have to be `&**obj` in order to go `&PyList` -> `PyList` (deref) -> ` PyAny`.

Maybe it's nicer to just recommend `.as_ref()` here for now.

Reply:
It's good to remind people they can call `PyAny` methods via Deref, which doesn't require any calling or `*`.  But to explicitly go to `PyAny`, `as_ref()` is probably better.